src/azure-cli-core/azure/cli/core/command_recommender.py

```diff
@@ -35,7 +35,55 @@ class AladdinUserFaultType(Enum):
     InvalidAccountName = 'InvalidAccountName'
 
 
-class CommandRecommender():
+def get_error_type(error_msg):
+    """The the error type of the failed command from the error message.
+    The error types are only consumed by aladdin service for better recommendations.
+    """
+
+    error_type = AladdinUserFaultType.Unknown
+    if not error_msg:
+        return error_type.value
+
+    error_msg = error_msg.lower()
+    if 'unrecognized' in error_msg:
+        error_type = AladdinUserFaultType.UnrecognizedArguments
+    elif 'expected one argument' in error_msg or 'expected at least one argument' in error_msg \
+            or 'value required' in error_msg:
+        error_type = AladdinUserFaultType.ExpectedArgument
+    elif 'misspelled' in error_msg:
+        error_type = AladdinUserFaultType.UnknownSubcommand
+    elif 'arguments are required' in error_msg or 'argument required' in error_msg:
+        error_type = AladdinUserFaultType.MissingRequiredParameters
+        if '_subcommand' in error_msg:
+            error_type = AladdinUserFaultType.MissingRequiredSubcommand
+        elif '_command_package' in error_msg:
+            error_type = AladdinUserFaultType.UnableToParseCommandInput
+    elif 'not found' in error_msg or 'could not be found' in error_msg \
+            or 'resource not found' in error_msg:
+        error_type = AladdinUserFaultType.AzureResourceNotFound
+        if 'storage_account' in error_msg or 'storage account' in error_msg:
+            error_type = AladdinUserFaultType.StorageAccountNotFound
+        elif 'resource_group' in error_msg or 'resource group' in error_msg:
+            error_type = AladdinUserFaultType.ResourceGroupNotFound
```

Comment:
Could we make the mapping relationship between these error messages and error codes into a dict, and then match the error types directly through the dict?
So we don't need to write so many judgment statements, and the code may be more readable
Such as:
```
error_type_dict = {
    'not found': AladdinUserFaultType.UnableToParseCommandInput,
    'misspelled': AladdinUserFaultType.UnknownSubcommand,
    'arguments are required': {
        '_subcommand': AladdinUserFaultType.MissingRequiredSubcommand,
        '_command_package': AladdinUserFaultType.UnableToParseCommandInput
    }
}
```

Reply:
It's a good idea! 

But i am afraid it may bring extra complexity. For example, the statements here need 15 key-value pairs?
```python
elif 'not found' in error_msg or 'could not be found' in error_msg \
            or 'resource not found' in error_msg:
        error_type = AladdinUserFaultType.AzureResourceNotFound
        if 'storage_account' in error_msg or 'storage account' in error_msg:
            error_type = AladdinUserFaultType.StorageAccountNotFound
        elif 'resource_group' in error_msg or 'resource group' in error_msg:
            error_type = AladdinUserFaultType.ResourceGroupNotFound
```
Besides, can we deal with the default logic? E.g, the ` error_type = AladdinUserFaultType.AzureResourceNotFound` above?

Reply:
Yes, this makes it easier to find and modify the error type corresponding to an error message

```
error_type_dict = {
    'not found': {
        "default error": AladdinUserFaultType.UnableToParseCommandInput,   // default error type
        "storage_account": AladdinUserFaultType.StorageAccountNotFound,
        "storage account": AladdinUserFaultType.StorageAccountNotFound,
        "resource_group": AladdinUserFaultType.ResourceGroupNotFound,
        "resource group": AladdinUserFaultType.ResourceGroupNotFound
    },
    ...
}
```

In fact, the essential difference between the two ways is that one is aggregated according to the error message and the other is aggregated according to the error type returned.
I personally think that we are more concerned about how to get the error type according to `error_msg` rather than what `error_msg` corresponds to an error type.